### PR TITLE
use npm instead of exec, and other improvements

### PR DESCRIPTION
**Pull request by LeMisterV**
Instead of using exec to run the hook command we use npm module.
This allows to have a better stdout experience (streamed, not dumped when all is finished).
It also helps to have the right PATH defined for hook process.

We also : 
- Improved install process so it can work for module itself
- added git-hook on module itself
- improved tests

### CODE REVIEW

**Comment by maxhoffmann** [comment]
Hey thanks for this PR! I’ll take a closer look at it later this week.

File: package.json [hunk 1]
```diff
@@ -25,8 +25,10 @@
   "author": "Maximilian Hoffmann",
   "license": "MIT",
   "devDependencies": {
-    "fs-extra": "^0.16.5",
     "tap-dot": "^1.0.0",
     "tape": "^3.5.0"
+  },
+  "dependencies": {
+    "npm": "^3.5.1"
```
Comment (maxhoffmann) [inline package.json]: I’m not sure installing `npm` locally is worth the benefit of having the correct `PATH` set up for the hooks. Do you know if this works without problems when the global and local `npm` version differ? Especially version 2 and 3 as they install packages differently.

**Comment by maxhoffmann** [comment]
Thanks for this PR! I think these are very good changes, especially streaming the output to the console.

My biggest concern is that this breaks when the global `npm` version is different to the local one. I’m not sure if this package should require `npm > 3.0.0`. At least this should be added to `README.md` then.

File: package.json (hunk 1 above)
Comment (maxhoffmann) [inline package.json]: Just tested it. This breaks if the global version is `< 3`.

**Comment by LeMisterV** [comment]
Hi,

I don't know how to handle this npm version problem....
I see some solutions, I don't know which is better :
- set a peer-dependency to npm
- add a npm link npm in the post-install script
- don't require npm, just exec it or spawn it... but this causes stdout problems.... I don't remember what...

I'll try to correct this quickly

**Comment by LeMisterV** [comment]
Hi,

I made some changes, so it now uses spawn to run npm scripts, so we don't have to install a local npm anymore... and it looks like it works the same...

**Comment by maxhoffmann** [comment]
Cool I will give it a try soon. Could you change the indentation back to 2 spaces please?

Can you tell me what `-s` does for `npm run`? It’s not mentioned running `npm help run`.

**Comment by LeMisterV** [comment]
the -s argument will tell npm to run silently. So you won't have something like this displayed :

> captain-git-hook@1.2.0 test /path/to/workspace/captain-git-hook
> NODE_ENV=testing tape test/ | tap-dot

just eye-candy and less verbose

I will change indentation asap

**Comment by maxhoffmann** [comment]
> the -s argument will tell npm to run silently.

Wow I was looking for this for a long time. :D Thanks a lot! :100: 

I’ll ignore the indentation in the tests and merge it. Thanks for helping out and improving this! :+1:

**Comment by maxhoffmann** [comment]
Unfortunately I had to revert this PR again. Spawn does not seem to work with Git clients like Sourcetree and Git Tower.

I’ll see if there is fix to this and merge it again, when there is a solution.

**Pull request merged**